### ISSUE TITLE
Verdaccio overwrites existing packages with the same version

### ISSUE THREAD

**Issue by ryanewtaylor** [verdaccio/verdaccio#875]
**Describe the bug**
Publishing a package with the same version as an existing package correctly responds with "Cannot publish over existing version" but a subsequent yarn install (from a clean cache) displays a hash collision error indicating the existing package served by Verdaccio was modified.

**To Reproduce**
1. Create a new package, @my-co/my-package
2. Modify package @my-co/my-package
    ```
    echo bump>>readme.md
    yarn version
    ```
3. Publish package @my-co/my-package
    ```
    npm pack
    npm publish
    ```
3. In another project, my-consumer, consume package @my-co/my-package
    ```
    yarn add @my-co/my-package
    yarn build
    ```
4. Modify original package, @my-co/my-package, but do not bump its version
    ```
    echo bump>>readme.md
    ```
5. Publish same versioned package, @my-co/my-package
    ```
    npm pack
    npm publish
    ```
    Here, we receive an expected error

        npm ERR! registry error parsing json
        npm ERR! code EPUBLISHCONFLICT
        npm ERR! publish fail Cannot publish over existing version.
        npm ERR! publish fail Update the 'version' field in package.json and try again.
        npm ERR! publish fail 
        npm ERR! publish fail To automatically increment version numbers, see:
        npm ERR! publish fail     npm help version

6. In consumer project, my-consumer, clear node_modules and yarn cache (so as to not pull from cache)
   ```
   rm nodule_modules -force -recursive
   yarn cache clean
   ```
7. Install packages in project, my-consumer
    ```
    yarn install
    ```
8. See error

        error [url-to-my-package-x.y.z.tgz]: Fetch succeeded for undefined. However, extracting "[url-to-my-package-x.y.z.tgz]" resulted in hash "some-hash", which did not match the requested hash "some-other-hash".
  

**Expected behavior**
`yarn install` should proceed without error. The package served by Verdaccio should not change when attempting to publish over an existing version (per the error in step 5).

**Configuration File (cat ~/.config/verdaccio/config.yaml)** 

```yaml
storage: ./storage
auth:
  activedirectory:
    url: "ldap://ad.myco.local"
    baseDN: 'dc=myco,dc=local'
    domainSuffix: 'myco.local'
packages:
  '@*/*':
    access: $all
    publish: $authenticated
  '**':
    access: $all
    publish: $authenticated
logs:
  - {type: stdout, format: pretty, level: http}
url_prefix: https://npmjs.myco.com

**Comment by ryanewtaylor** [comment]
@juanpicado @dlouzan , I checked this issue against #874 and they appear to be describing the same issue. I believe the package-x.y.z.tgz file may also be replaced in the file system too as the last modified date changes despite the error.

**Comment by smelukov** [comment]
The same issue

**Comment by juanpicado** [comment]
Released on `v3.6.0`
